ludwig/automl/auto_tune_config.py

```diff
@@ -0,0 +1,164 @@
+    'tabnet': OrderedDict({
+        'training.batch_size': 32,
+        'combiner.size': 8,
+        'combiner.output_size': 8,
+    }),
+    'concat': OrderedDict({
+        'training.batch_size': 32,
+        'combiner.fc_size': 64,
+        'combiner.num_fc_layers': 1,
+
+    }),
+    'tabtransformer': OrderedDict({
+        'training.batch_size': 32,
+        'combiner.num_heads:': 4,
+        'combiner.output_size': 8,
+        'combiner.num_layers': 4,
+        'combiner.num_fc_layers': 1,
+    }),
+}
+
+
+BYTES_TO_MB = 1e6
+
+
+def get_trainingset_metadata(config, dataset):
+    (_, _, _, training_set_metadata) = preprocess_for_training(
+        config,
+        dataset=dataset,
+        preprocessing_params=config[PREPROCESSING])
+    return training_set_metadata
+
+
+def get_machine_memory():
+
+    if ray.is_initialized():  # using ray cluster
+        @ray.remote(num_gpus=1)
+        def get_remote_gpu():
+            gpus = GPUtil.getGPUs()
+            total_mem = gpus[0].memory_total
+            return total_mem * BYTES_TO_MB
+
+        @ray.remote(num_cpus=1)
+        def get_remote_cpu():
+            total_mem = psutil.virtual_memory().total
+            return total_mem
+
+        resources = get_available_resources()  # check if cluster has GPUS
+
+        if resources['gpu'] > 0:
+            machine_mem = ray.get(get_remote_gpu.remote())
+        else:
+            machine_mem = ray.get(get_remote_cpu.remote())
+    else:  # not using ray cluster
+        if GPUtil.getGPUs():
+            machine_mem = GPUtil.getGPUs()[0].memory_total * BYTES_TO_MB
+        else:
+            machine_mem = psutil.virtual_memory().total
+
+    return machine_mem
+
+
+def compute_memory_usage(config, training_set_metadata) -> int:
+    update_config_with_metadata(config, training_set_metadata)
+    lm = LudwigModel.create_model(config)
+    lm.get_connected_model()
+    model_tensors = lm.collect_weights()
+    total_size = 0
+    batch_size = config[TRAINING][BATCH_SIZE]
+    for tnsr in model_tensors:
+        total_size += tnsr[1].numpy().size * batch_size
+    total_bytes = total_size * 32  # assumes 32-bit precision
+    return total_bytes
+
+
+def sub_new_params(config: dict, new_param_vals: dict):
+    new_config = copy.deepcopy(config)
+    for param, val in new_param_vals.items():
+        config_section = param.split(".")[0]
+        param_name = param.split(".")[1]
+        new_config[config_section][param_name] = val
+    return new_config
+
+
+def get_new_params(current_param_values, hyperparam_search_space, params_to_modify):
+    for param, _ in params_to_modify.items():
+        if hyperparam_search_space[param][SPACE] == "choice":
+            current_param_values[param] = hyperparam_search_space[param]['categories'][-1]
+        else:
+            current_param_values[param] = hyperparam_search_space[param]['upper']
+    return current_param_values
+
+
+def memory_tune_config(config, dataset):
```

Comment:
If would suggest running this with `ray.remote` if ray is available, just because TensorFlow has a tendency to have some issues with state corruption when creating models and potentially allocating GPU memory.